src/store/modules/accounts/hdWallet.js

```diff
@@ -33,11 +40,31 @@ export default {
     },
     async discover({ state, rootGetters, dispatch }) {
       let lastNotEmptyIdx = 0;
-      let account;
-      for (let nextIdx = state.nextAccountIdx; nextIdx <= 5; nextIdx += 1) {
-        account = getHdWalletAccount(rootGetters.wallet, nextIdx);
-        // eslint-disable-next-line no-await-in-loop
-        if (await dispatch('isAccountUsed', account.address)) lastNotEmptyIdx = nextIdx;
+      let lastIndex = 0;
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
Infinite loops must be avoided. I feel this logic is not easy to understand. Maybe you can simplify or add comments